packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -290,61 +296,64 @@ async function uploadFile(
     'uploadConcurrency',
     uploadOptions.uploadConcurrency
   )
-  const maxChunkSize = utils.assertDefined(
-    'uploadChunkSize',
-    uploadOptions.uploadChunkSize
-  )
+  const maxChunkSize = 32 * 1024 * 1024 // 32 MB Chunks
```

Comment:
Should we dynamically calculate the chunk size depending on the file size? Also, looks like we upload same size of file chinks until the very last item but are you sure that works? Seems like there are some issues in R2 with different size of file chunk uploads 
- https://github.com/apache/arrow/issues/34363
- https://github.com/apache/arrow-rs/issues/4304

Reply:
The implementation seems to upload chunks of exactly 32 MB except for the last chunk, which would be <= 32 MB. That seems to comply to R2's limitations:

[​​Limitations](https://developers.cloudflare.com/r2/objects/multipart-objects/#limitations)
Object part sizes must be at least 5MiB but no larger than 5GiB. All parts except the last one must be the same size. The last part has no minimum size, but must be the same or smaller than the other parts.

Reply:
The GitHub cache client includes a configuration value for setting the maximum chunk size, which defaults to 32MB. It remains unchanged in most customer workflows. Given the varying multipart upload workflows of Azure Blob Storage and S3 compatible storages, the server also needs to be aware of this maximum chunk size in S3 case.

I believe there's no need for a more complex chunk size calculation as it's also fixed for the default GitHub cache. However, I've added the maximum chunk size value to reserveCache response, enabling the backend to determine it. This logic can be enhanced in the future without causing any disruption to customers.